builtin/providers/vsphere/resource_vsphere_file.go

```diff
@@ -0,0 +1,253 @@
+package vsphere
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"golang.org/x/net/context"
+)
+
+type file struct {
+	datacenter      string
+	datastore       string
+	sourceFile      string
+	destinationFile string
+}
+
+func resourceVSphereFile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereFileCreate,
+		Read:   resourceVSphereFileRead,
+		Update: resourceVSphereFileUpdate,
+		Delete: resourceVSphereFileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"datacenter": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"datastore": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"source_file": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"destination_file": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceVSphereFileCreate(d *schema.ResourceData, meta interface{}) error {
+
+	log.Printf("[DEBUG] creating file: %#v", d)
+	client := meta.(*govmomi.Client)
+
+	f := file{
```

Comment:
What will happen if any of these are omitted? Will vsphere handle empty/null values?

I think this will need to be handled in the same way as datacenter